### PR TITLE
Shatz batch sampler

### PR DESCRIPTION
**Pull request by smartdanny**
Working batch sampler in ddp mnist example :)

### CODE REVIEW

**Comment by smartdanny** [comment]
I also cleaned up the mnist example a bit. Added a datamodule (seems necessary for ddp, but also I think its good style) and no model saving/checkpointing anymore (I think better for debugging).

**Review by mosheraboh — APPROVED** [review]
Looks great! Thanks @smartdanny 
Please see my questions inline.
And note that the mnist test currently fails and should be modified as well.

File: fuse_examples/imaging/classification/mnist/simple_mnist_starter.py [hunk 1]
```diff
@@ -125,52 +117,82 @@ def __init__(self, model_dir: str, lr: float, weight_decay: float):  # paths, tr
...
     )
 
     # create pl trainer
     pl_trainer = pl.Trainer(
-        default_root_dir=paths["model_dir"],
-        max_epochs=train_params["trainer.num_epochs"],
-        accelerator=train_params["trainer.accelerator"],
-        devices=train_params["trainer.num_devices"],
+        # default_root_dir=params["model_dir"],
+        replace_sampler_ddp=False,  # Must be set when using a batch sampler
```
[the remark this answers is not in the document]
Comment (smartdanny) [reply fuse_examples/imaging/classification/mnist/simple_mnist_starter.py:186]: Will add :)

File: fuse_examples/imaging/classification/mnist/simple_mnist_starter.py [hunk 2]
```diff
@@ -125,52 +117,82 @@ def __init__(self, model_dir: str, lr: float, weight_decay: float):  # paths, tr
...
+        distributed=distributed,
     )
 
     # create pl trainer
     pl_trainer = pl.Trainer(
-        default_root_dir=paths["model_dir"],
-        max_epochs=train_params["trainer.num_epochs"],
-        accelerator=train_params["trainer.accelerator"],
-        devices=train_params["trainer.num_devices"],
+        # default_root_dir=params["model_dir"],
```
[the remark this answers is not in the document]
Comment (smartdanny) [reply fuse_examples/imaging/classification/mnist/simple_mnist_starter.py:185]: I removed automatic saving/checkpointing of anything entirely. Just for more simplicity in the mnist example.

[the remark this answers is not in the document]
Comment (smartdanny) [reply fuse_examples/imaging/classification/mnist/simple_mnist_starter.py:185]: Ill delete this line now

File: fuse_examples/imaging/classification/mnist/simple_mnist_starter.py [hunk 3]
```diff
@@ -50,18 +47,12 @@
...
+PARAMS_DICT = {
+    "paths.data_dir": "_examples/mnist",
     "data.batch_size": 100,
-    "data.train_num_workers": 8,
-    "data.validation_num_workers": 8,
-    "trainer.num_epochs": 2,
-    "trainer.num_devices": 1,
+    "data.num_workers": 8,
+    "trainer.num_epochs": 3,
+    "trainer.num_devices": 2,
```
[the remark this answers is not in the document]
Comment (smartdanny) [reply fuse_examples/imaging/classification/mnist/simple_mnist_starter.py:55]: Yea, sorry forgot to change back. Will do now.

**Review by mosheraboh — APPROVED** [review]

**Pull request merged**